portfolio/src/main/webapp/script.js

```diff
@@ -12,12 +12,115 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+var context;
+var canvas;
+var myBubbleArray;
+var numBubs = 20;
+var animationID = null;
+// Resizes bubbles when page size is changed
+window.addEventListener('resize', init)
+
+function startChat() {
+  grabComments();
+  init();
+  grabUser();
+}
+
+class Bubble {
+  constructor(color) {
+    this.speed_factor = 3;
+    if (canvas.width < 900) {
+      this.speed_factor = 1.75;
+    }
+    this.color = color;
+    this.size = Math.random() * canvas.width;
+    this.x = Math.random() * canvas.width;
+    this.y = Math.random() * canvas.height;
+    this.radius = Math.floor(Math.random() * Math.sqrt(canvas.width));
+    this.dx = Math.random() * this.speed_factor * this.randomDirection();
+    this.dy = Math.random() * this.speed_factor * this.randomDirection();
+    this.opacity = Math.random();
+  }
+  drawBubble() {
```

Comment:
Like above, drawBubble should not rely on the global context variable; instead it should retrieve the context from the stored canvas instance variable.

That way we reduce reliance on globals and we also ensure that the context we draw on actually belongs to the original canvas used to create the bubble.